jax/experimental/mesh_utils.py

```diff
@@ -143,6 +179,7 @@ def _jax_devices_order_normalized(
   # h_x, h_y can be 2x2 or 1x1 depending on tasks_per_host=4 or 1
   h_x, h_y, _, cores_per_chip = local_topology
   assert cores_per_chip == 1
+  assert (h_x, h_y) in [(1, 1), (2, 2)]
```

Comment:
can you remove this assertion? certain system requires more generality here

Reply:
Done.